spotbugs/src/main/java/edu/umd/cs/findbugs/detect/AbstractAssertDetector.java

```diff
@@ -0,0 +1,51 @@
+package edu.umd.cs.findbugs.detect;
+
+import org.apache.bcel.Const;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+
+/**
+ * Abstract base class for finding assertions
```

Comment:
Why is an abstract base class needed?

Reply:
Because it is the common ancestor of this detector and #2130.